runtime/jvmti/jvmtiThread.cpp

```diff
@@ -1179,11 +1179,27 @@ resumeThread(J9VMThread *currentThread, jthread thread)
 #if JAVA_SPEC_VERSION >= 19
 		J9JavaVM *vm = currentThread->javaVM;
 		j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
+		U_64 isSuspendedFlag = J9OBJECT_U64_LOAD(currentThread, threadObject, vm->isSuspendedInternalOffset);
+		BOOLEAN inTransition = J9_ARE_ANY_BITS_SET(isSuspendedFlag, ~0x1);
+
+		/* Update internal flag before clearing flag to avoid sync issue. */
+		if (J9_ARE_NO_BITS_SET(J9OBJECT_U64_LOAD(currentThread, threadObject, vm->isSuspendedInternalOffset), 1)) {
+			rc = JVMTI_ERROR_THREAD_NOT_SUSPENDED;
+		} else {
+			J9OBJECT_U64_STORE(currentThread, threadObject, vm->isSuspendedInternalOffset, (isSuspendedFlag & ~0x1));
+			if (inTransition) {
+				targetThread = (J9VMThread*)(isSuspendedFlag & ~0x1);
+				/* For virtual thread in transition, ensure thread cannot start running until releaseVMThread is called. */
+				vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
```

Comment:
since the target thread is already halted by suspend flag, `haltThreadForInspection` should directly return after setting the inspection flag, though i guess it wouldn't hurt to do the refetch.